TYPO3.Neos/Classes/TYPO3/Neos/Domain/Model/Site.php

```diff
@@ -279,17 +279,17 @@ public function getFirstActiveDomain()
     /**
      * Sets (and adds if necessary) the primary domain of this site.
      *
-     * @param Domain $domain The domain
+     * @param Domain|null $domain The domain
      * @return void
      * @api
      */
-    public function setPrimaryDomain(Domain $domain)
+    public function setPrimaryDomain(Domain $domain = null)
     {
-        if (!$domain->getActive()) {
+        if ($domain != null && !$domain->getActive()) {
```

Comment:
Also I guess this could be overally easier if you first check for `$domain === null` and just set null and return. That means after that first if you can assume that $domain is actually a Domain object.